### PR TITLE
Scoring and Analysis of Adapter Experiments

### PR DESCRIPTION
**Pull request by shan18**
# What does this PR do ?

Added the script for scoring and analysis of the adapter experiments.

**Collection**: asr

# Changelog 
- Created a new script: `examples/asr/asr_adapters/scoring_and_analysis.py`
- Updated README: `examples/asr/asr_adapters/README.md`

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

## Who can review?

Anyone in the NeMo ASR Team

### CODE REVIEW

File: examples/asr/asr_adapters/README.md [hunk 1]
```diff
@@ -58,3 +58,51 @@ graph TD
 ```
 
 **Note**: If you with to evaluate the base model (with all adapters disabled), simply pass `model.adapter.adapter_name=null` to the config of this script to disable all adapters and evaluate just the base model.
+
+# Scoring and Analysis of Results
```
Comment (titu1994) [inline examples/asr/asr_adapters/README.md:62]: This basic usage documentation should be inside of the scoring and analysis.py script probably.

File: examples/asr/asr_adapters/scoring_and_analysis.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This script is used to analyze the results of the experiments from a CSV file.
+
+Usage:
+    python scoring_and_analysis.py \
+        --csv <path to cleaned result csv file> \
+        --dataset_type_column <column in csv with the dataset types>
```
Comment (titu1994) [inline examples/asr/asr_adapters/scoring_and_analysis.py:21]: Move the basic usage documentation inside this script.

File: examples/asr/asr_adapters/scoring_and_analysis.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+"""
+
+import argparse
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+
+# Column name and its attribute to consider for getting the best model configuration
+UNCONSTRAINED_EXP_KEY = {'name': 'WER: Test', 'attribute': min}
```
Comment (titu1994) [inline examples/asr/asr_adapters/scoring_and_analysis.py:31]: Note down in comments which of these arguments users must change.

File: examples/asr/transcribe_speech.py [hunk 4]
```diff
@@ -172,8 +172,9 @@ def main(cfg: TranscriptionConfig) -> TranscriptionConfig:
     if hasattr(asr_model, 'change_decoding_strategy'):
         # Check if ctc or rnnt model
         if hasattr(asr_model, 'joint'):  # RNNT model
-            rnnt_decoding = RNNTDecodingConfig(fused_batch_size=-1, compute_langs=cfg.compute_langs)
-            asr_model.change_decoding_strategy(rnnt_decoding)
+            cfg.rnnt_decoding.fused_batch_size = -1
```
Comment (titu1994) [inline examples/asr/transcribe_speech.py:175]: Rebase your pr, I've merged the other pr with these changes.

**Review by titu1994 — COMMENTED** [review]
Overall looks good, requires a bit of shuffling of the docstring + more comments on which args users must change and what should be format of the csv file c

File: examples/asr/transcribe_speech.py (hunk 4 above)
Comment (shan18) [reply examples/asr/transcribe_speech.py]: This is done

File: examples/asr/asr_adapters/README.md (hunk 1 above)
Comment (shan18) [reply examples/asr/asr_adapters/README.md:62]: This is done.

For now, the basic usage documentation is present in the README as well as inside the script. Should I remove it from the README?

File: examples/asr/asr_adapters/scoring_and_analysis.py (hunk 2 above)
Comment (shan18) [reply examples/asr/asr_adapters/scoring_and_analysis.py]: Done

**Review by titu1994 — DISMISSED** [review]
Looks great ! Thanks !

**Review by titu1994 — APPROVED** [review]

**Pull request merged**